glib-macros/src/thread_local_object.rs

```diff
@@ -0,0 +1,72 @@
+// Take a look at the license at the top of the repository in the LICENSE file.
+
+use proc_macro2::TokenStream;
+use quote::{quote, quote_spanned, ToTokens};
+use syn::{
+    parse::{Parse, ParseStream},
+    spanned::Spanned,
+    token, Expr, Ident, TypePath,
+};
+
+pub(crate) struct ThreadLocalObjectTokens {
+    name: Ident,
+    ty: TypePath,
+    init_expr: Option<Expr>,
+}
+
+impl Parse for ThreadLocalObjectTokens {
+    fn parse(input: ParseStream) -> syn::Result<Self> {
+        let name: Ident = input.parse()?;
+        let _comma: token::Comma = input.parse()?;
+        let ty: TypePath = input.parse()?;
+
+        let init_expr = if input.is_empty() {
+            None
+        } else {
+            let _comma: token::Comma = input.parse()?;
+            Some(input.parse()?)
+        };
+
+        Ok(Self {
+            name,
+            ty,
+            init_expr,
+        })
+    }
+}
+
+impl ToTokens for ThreadLocalObjectTokens {
```

Comment:
This could also just be a declarative macro in the glib crate or not? :) Having it as a proc-macro just complicates things